src/main/java/com/example/booksstore/service/impl/BookServiceImpl.java

```diff
@@ -36,4 +36,19 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto book) {
+        if (bookRepository.existsById(id)) {
+            Book updatedBook = bookMapper.toModel(book);
+            updatedBook.setId(id);
+            return bookMapper.toDto(bookRepository.save(updatedBook));
+        }
+        throw new EntityNotFoundException("Can't find book by id " + id);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
```

Comment:
we should also check if that book exists first, and be careful to not duplicate the code